src/main/java/com/takipi/oss/benchmarks/jmh/loops/ByteArrayLoopBenchmark.java

```diff
@@ -0,0 +1,107 @@
+package com.takipi.oss.benchmarks.jmh.loops;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(2)
+@Measurement(iterations = 5)
+@Warmup(iterations = 5)
+public class ByteArrayLoopBenchmark {
+    int size = 100_000;
+
+    int byteArrLength = 100;
+
+    List<byte[]> arrays;
+
+    @Setup
+    public void setup() {
+        arrays = new ArrayList<>(size);
+
+        Random random = new Random();
```

Comment:
Compare "max" algorithm for this data sets :)

1 2 3 4 5 6 7 8 9 10
1 1 1 1 1 1 1 1 1 1
10 9 8 7 6 5 4 3 2 1
3 2 5 1 2 6 3 8 7 6 5


Now imagine that each benchmark case will get it's own random sequence. And sequence for one could be much simpler compared to another